app/models/match_decisions/provider_only/hsa_acknowledges_receipt.rb

```diff
@@ -77,9 +77,9 @@ def acknowledged
         end
       end
 
-      def canceled
+      def canceled contact
         Notifications::MatchCanceled.create_for_match! match
-        match.canceled!
+        match.canceled! contact&.id
```

Comment:
`ClientOpportunityMatch#canceled!` takes a named parameter, so this won't work as written.